.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,48 @@
+# This action script will make sure that project gets build and checked on every commit
+# and on pull request
+
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches-ignore: [master]
```

Comment:
Does this mean that it doesn't build when commits are pushed to `master`? I think it should.

Reply:
Yes this means if we directly push to master it won't run
Again ideally we shouldn't commit directly to master it should always go via PR, hence line no. 11 and 12

Reply:
In my opinion it should also build when commits are directly pushed to `master`. Allows us to spot failures earlier rather than later.

Generally it's good to have the changes coming through pull requests, but probably this isn't always the case. It's good to be able to do minor changes directly on the `master`.